File: StemExplorerAPI/StemExplorerAPI/Models/Entities/ChallengeCompact.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StemExplorerAPI.Models.Entities
{
    public class ChallengeCompact
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public Enums.ChallengeCategories Category { get; set; }
        public string Question { get; set; }
        public string Instructions { get; set; }
        public string Hint { get; set; }
        public string PossibleAnswers { get; set; } // json string
        public int Answer { get; set; }
        public string AnswerBlurb { get; set; }
        public bool IsComplete { get; set; }
    }
}

File: stem-explorer-ng/src/challenge/models/challenge-compact.ts
import { StemCategory } from '../../app/shared/enums/stem-cateogry.enum';

export interface ChallengeCompact {
  id: number;
  title: string;
  category: StemCategory;
  question: string;
  instructions: string;
  hint: string;
  possibleAnswers: { idx: number; label: string }[];
  answer: number;
  answerBlurb: string;
  isComplete: boolean;
}
